doc/wiki/Index.md

```diff
@@ -35,6 +35,7 @@
     - [Helpful Compiler Flags](testing/Helpful_Compiler_Flags_For_Testing.md)
 
 - [Design Documentation](design/Design_Documentation_Index.md)
+- [Developer Guide](developer_guide.md)
```

Comment:
The link was already added in "Contributing to MPICH"

Reply:
No. "Design Documentation" currently is a catch bag.

Reply:
Removed this commit.